protocol/x/clob/keeper/order_state.go

```diff
@@ -125,14 +128,32 @@ func (k Keeper) GetOrderFillAmount(
 	return true, satypes.BaseQuantums(orderFillState.FillAmount), orderFillState.PrunableBlockHeight
 }
 
-// AddOrdersForPruning creates or updates a slice of `orderIds` to state for potential future pruning from state.
-// These orders will be checked for pruning from state at `prunableBlockHeight`. If the `orderIds` slice provided
-// contains duplicates, the duplicates will be ignored.
+func (k Keeper) GetPruneableOrdersStore(ctx sdk.Context, height uint32) prefix.Store {
+	var buf bytes.Buffer
+	buf.Write([]byte(types.PrunableOrdersKeyPrefix))
+	buf.Write(lib.Uint32ToKey(height))
+	buf.Write([]byte(":"))
+	return prefix.NewStore(ctx.KVStore(k.storeKey), buf.Bytes())
+}
+
+// AddOrdersForPruning creates or updates `orderIds` to state for potential future pruning from state.
 func (k Keeper) AddOrdersForPruning(ctx sdk.Context, orderIds []types.OrderId, prunableBlockHeight uint32) {
+	store := k.GetPruneableOrdersStore(ctx, prunableBlockHeight)
+	for _, orderId := range orderIds {
+		store.Set(
```

Comment:
We're still using the proto serialized version of orderid as the state key?

What is the difference between these two marshals?
`orderId.ToStateKey(),` `k.cdc.MustMarshal(&orderId),`

Reply:
Yeah. I will change that for the whole codebase later. For this specific change it doesn't really matter since we are iterating over prefix rather than trying to query keys directly.

Reply:
Right now those two marshals are the same, but it will be different later.